src/test/java/com/example/music_player/storage/CloudStorageAmazonS3Test.java

```diff
@@ -67,21 +78,43 @@ public void saveWhenBucketNotExist() {
 
         verify(s3Client, times(1)).createBucket(bucketName);
         verify(s3Client, times(1)).putObject(any(PutObjectRequest.class));
-        assertThat(fileObject.delete()).isEqualTo(false);
     }
 
     @Test
-    public void saveWhenBucketExistNow() {  //TODO Nullpointer s3Client in CloudStorageAmazonS3, How to properly mock it?
+    public void saveWhenBucketExistNow() {
         when(s3Client.doesBucketExistV2(bucketName)).thenReturn(true);
         when(s3Client.putObject(any())).thenReturn(putObjectResult);
 
         cloudStorageAmazonS3.save(inputStream, "originalFilename", "contentType");
 
         verify(s3Client, never()).createBucket(bucketName);
         verify(s3Client, times(1)).putObject(any(PutObjectRequest.class));
-        assertThat(fileObject.delete()).isEqualTo(false);
     }
 
+    @Test
+    public void isTempFileDeletedWhenThrowException() throws IOException {
+
+        try (MockedStatic<File> utilities = Mockito.mockStatic(File.class)) {
+            utilities.when(() -> File.createTempFile(anyString(), anyString())).thenReturn(testedFile);
+            assertThat(File.createTempFile("Epam_MusicPlayer-", ".tmp")).isEqualTo(testedFile);
+            assertEquals(testedFile.length(), 0);
+        }
+
+        when(s3Client.putObject(any())).thenThrow(new RuntimeException());
+
+        try (MockedStatic<Collections> utilities2 = Mockito.mockStatic(Collections.class)) {
+            utilities2.when(() -> Collections.singletonList(any(Source.class)))
+                    .thenReturn(sourceList);
+        }
+//        assertThat(cloudStorageAmazonS3.save(inputStream, "originalFilename", "contentType")
+//                .get(0)
+//                .getStorage_types())
+//                .isEqualTo("CLOUD_STORAGE");
+       cloudStorageAmazonS3.save(inputStream, "originalFilename", "contentType");
+        assertThat(testedFile.exists()).isFalse();
```

Comment:
Mocked testedFile.exists() always returns false because it is default result for mock methods thar were not overwritten by Mockito.when(), so this assert doesn't bring any actual verification. Instead you should verify that delete() method is invoked at least once.

Reply:
in this key, will be use Power Mokito